packages/devtools_app/lib/src/app.dart

```diff
@@ -443,6 +443,15 @@ class DevToolsScreen<C extends Object?> {
   ///
   /// If [createController] and [controller] are both null, [screen] will be
   /// responsible for creating and maintaining its own controller.
+  ///
+  /// The controller should expect to be already connected to an application
+  /// at the time of creation.
+  /// But it should invoke unawaitable `serviceConnection.serviceManager.onServiceAvailable`
+  /// in the constructor to make sure the service is ready to be used.
```

Comment:
In the controller initialization, if logic requires a connected [VmService] object (`serviceConnection.serviceManager.service`), then the controller should first await
the `serviceConnection.serviceManager.onServiceAvailable` future to ensure the service has
been initialized.